src/Commands/MeltorCommand.php

```diff
@@ -0,0 +1,309 @@
+                    } else {
+                        $columnMigrations[] = $columnMigration;
+
+                        if (str_contains($columnMigration, '->increments(') || str_contains($columnMigration, '->id(')) {
+                            $this->primaryKeyIsSetFor[$tableName] = true;
+                        }
+                    }
+                }
+            }
+
+            // Unique Keys.
+            foreach ($uniqueKeys->get($tableName) ?? [] as $uniqueKey) {
+                $columnMigrations[] = $this->meltor->generateUniqueKeyMigration($uniqueKey, $this->problems['nonGeneratedColumns'] ?? [], $this);
+            }
+
+            // Index Keys.
+            foreach ($this->meltor->getIndexesFor($tableName, $this->canAccessInnoDbIndexes, $this->indexes, $this->dataConnection, true) as $indexKeyName => $indexKeyColumns) {
+                $columnMigrations[] = $this->meltor->generateIndexKeyMigration($indexKeyName, $indexKeyColumns);
+            }
+
+            // Primary Key
+            if (!($this->primaryKeyIsSetFor[$tableName] ?? false) && $primaryKey = $this->meltor->getPrimaryKeyFor($tableName, $this->dataConnection)) {
+                $columnMigrations[] = $this->meltor->generatePrimaryKeyMigration($primaryKey);
+            }
+
+            // Spatial Keys.
+            foreach (
+                $this->meltor->getIndexesFor(
+                    $tableName,
+                    $this->canAccessInnoDbIndexes,
+                    $this->spatialIndexes,
+                    $this->dataConnection,
+                    false
+                ) as $indexKeyName => $indexKeyColumns
+            ) {
+                $columnMigrations[] = $this->meltor->generateIndexKeyMigration($indexKeyName, $indexKeyColumns, true);
+            }
+
+            // Foreign Keys.
+            foreach ($foreignKeys->get($tableName)?->groupBy('CONSTRAINT_NAME') ?? [] as $foreignKey) {
+                $foreignKeyMigration = $this->meltor->generateForeignKeyMigration($foreignKey, $this->ignoreProblems, $this);
+                if ($this->separateForeignKeys) {
+                    $columnConstraintMigrations[] = $foreignKeyMigration;
+                } else {
+                    $columnMigrations[] = $foreignKeyMigration;
+                }
+            }
+
+            $tableMigrations[$tableName]['laravel'] = $columnMigrations;
+            $tableMigrations[$tableName]['raw']     = $columnMigrationsRaw;
+
+            if ($this->separateForeignKeys) {
+                $constraintMigrations[$tableName] = $columnConstraintMigrations;
+            }
+        }
+
+        $migrationCode      = $this->meltor->generateMigration($tableMigrations, $this->meltor->getMigrationTemplate('migrationComment'), $constraintMigrations, $rawExceptions);
+        $migrationCode      = $this->meltor->beautify($migrationCode);
```

Comment:
It would be nice if meltor kept the migration code within the instance and all those methods would return a meltor instance so you could chain the calls and don't have to pass the migration code in and out each time. I guess the same goes for all the other data... However, I guess you made that design decision for a reason.